src/test/java/org/takes/rs/RsWithTypeTest.java

```diff
@@ -201,21 +256,77 @@ public void replacesTypeWithXml() throws Exception {
 
     /**
      * RsWithType.Text can replace an existing type with text/plain.
-     * @throws java.lang.Exception If a problem occurs.
+     * @throws Exception If a problem occurs.
      */
     @Test
     public void replacesTypeWithText() throws Exception {
         MatcherAssert.assertThat(
             new RsPrint(
+                new RsWithType.Text(new RsWithType(new RsEmpty(), TYPE_HTML))
+            ).print(),
+            Matchers.equalTo(
+                Joiner.on(CRLF).join(
+                    HTTP_OK,
+                    String.format(CONTENT_TYPE, TYPE_TEXT),
+                    "",
+                    ""
+                )
+            )
+        );
+        MatcherAssert.assertThat(
+            new RsPrint(
                 new RsWithType.Text(
-                    new RsWithType(
-                        new RsEmpty(), TYPE_HTML
-                    )
+                    new RsWithType(new RsEmpty(), TYPE_HTML),
+                    StandardCharsets.ISO_8859_1
+                )
+            ).print(),
+            Matchers.equalTo(
+                Joiner.on(CRLF).join(
+                    HTTP_OK,
+                    String.format(
+                        CONTENT_TYPE_WITH_CHARSET,
+                        TYPE_TEXT,
+                        StandardCharsets.ISO_8859_1
+                    ),
+                    "",
+                    ""
+                )
+            )
+        );
+    }
+
+    /**
+     * RsWithType can add the charset to the content type when it is explicitly
+     * specified and can still provide the content type without the charset
+     * when it is not explicitly specified.
```

Comment:
@essobedo are you sure that we need one test to test two things (content with charset and without charset)?